src/CurlParser.js

```diff
@@ -3,526 +3,529 @@ import Immutable from 'immutable'
+            }
+            else {
+                cleanedArgs.push(arg)
+            }
+        })
+        args = Immutable.List(cleanedArgs)
+
+        return args
     }
-    return request.setIn(['headers', this._normalizeHeader(m[1])], m[2])
-  }
 
-  _parseMethod(request) {
-    return request.set('method', this._popArg())
-  }
+    parse(string) {
+        // tokenize
+        this.args = this._tokenize(string)
 
-  _parseCompressed(request) {
-    let acceptEncoding = request.getIn(['headers', 'Accept-Encoding'])
-    if (!acceptEncoding) {
-      acceptEncoding = ''
+        // parse
+        this.idx = 0
+        this._parseAll()
+
+        return this.requests
     }
-    if (acceptEncoding.indexOf('gzip') < 0) {
-      if (acceptEncoding.length > 0) {
-        acceptEncoding += ';'
-      }
-      acceptEncoding += 'gzip'
-      request = request.setIn(['headers', 'Accept-Encoding'], acceptEncoding)
+
+    _parseAll() {
+        let arg
+        while ((arg = this._popArg()) != null) {
+            if (arg.toLowerCase() === 'curl') {
+                ::this._parseCurlCommand()
+
+                // if last argument was the -: --next option, continue parsing curl
+                const lastArg = this._getLastArg()
+                if (lastArg === '-:' || lastArg === '--next') {
+                    ::this._parseCurlCommand()
+                }
+            }
+        }
     }
-    return request
-  }
-
-  _parseUserAgent(request) {
-    return request.setIn(['headers', 'User-Agent'], this._popArg())
-  }
-
-  _parseCookie(request) {
-    return request.setIn(['headers', 'Cookie'], this._popArg())
-  }
-
-  _parseReferer(request) {
-    // note: spelling "referer" is a typo in the HTTP spec
-    // while correct English is "Referrer", header name if "Referer" (one R)
-    return request.setIn(['headers', 'Referer'], this._popArg())
-  }
-
-  _parseUser(request) {
-    const m = this._popArg().match(/([^\:]+)(?:\:([\s\S]*))?/)
-    return request.setAuthParams({
-      username:m[1],
-      password:(m[2] ? m[2] : null)
-    })
-  }
-
-  _parseAuth(request, arg) {
-    const m = arg.match(/\-{2}(\w+)/)
-    return request.setAuthParams({
-      type:m[1]
-    })
-  }
-
-  _parseMaxTime(request) {
-    const maxTime = this._popArg()
-    return request.set('timeout', parseFloat(maxTime))
-  }
-
-  _parseMultipartFormData(request) {
-    // switch bodyType
-    if (request.get('bodyType') != 'formData') {
-      if (request.get('bodyType') != null) {
-        throw new Error('Different body types set in the same request')
-      }
-      request = request.merge({
-        bodyType: 'formData',
-        body: Immutable.OrderedMap()
-      })
+
+    _parseCurlCommand() {
+        let request = new CurlRequest()
+        let urls = Immutable.List()
+        let arg
+        while ((arg = this._popArg()) != null) {
+            if (arg.toLowerCase() === 'curl') {
+                this._parseCurlCommand()
+            }
```

Comment:
Before, we were not checking for the `curl` parameter in our conditions, therefore it fell in the final else (line:283), which generated a url from which it would create a request.

Therefore, the following commands:
```sh
curl -X "GET" "https://echo.luckymarmot.com" \
    -H "abc: bab"
curl -X "GET" "https://echo.luckymarmot.com" \
    -H "abc: bcb"
```

would yield 3 requests with the following urls:
- https://echo.luckymarmot.com
- curl
- https://echo.luckymarmot.com

There's probably a better way to fix this, but it passes tests and since we're moving away from cURL-Importer in favor of API-Flow anyway, I don't really see the point of improving it more than that